src/Modules/Authontication/Components/LogIn/LogIn.tsx

```diff
@@ -1,7 +1,144 @@
-import React from 'react'
+import { useForm } from 'react-hook-form';
+
+import { useContext, useState } from 'react';
+import { Link, useNavigate } from 'react-router-dom';
+import axios from 'axios';
+import { toast } from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.css';
+import { USERS_URLs } from '../../../../constans/END_POINTS';
+import { AuthContext } from '../Context/AuthContext'
+import { EmailValidation, PasswordValidation } from '../../../../constans/VALIDATIONS';
 
 export default function LogIn() {
+  let { saveLoginData } = useContext(AuthContext);
+  let navigate = useNavigate();
+  let{
+    register,
+    handleSubmit,
+    formState:{errors},
+  } = useForm({ mode: "onBlur" });
+
+
+  const [isPasswordVisible, setIsPasswordVisible] = useState(false);
+  const toggleVisibility = (setterFunction: any) => {
+    setterFunction((prevState: any) => !prevState);
+  };
+
+  let onSubmit = async (data:any)=>{
+    try {
+      let response = await axios.post(USERS_URLs.Login, data);
+      localStorage.setItem('token', response.data.token);
+      saveLoginData();
```

Comment:
Here as I mentioned, we can use the token  to call the currentUser api and get the full user data instead of using the jwt-decode library.

Reply:
do you mean we can call this API : /api/v1/Users/currentUser  ?
without sending any parameters, just being logged in the app ? like this
![Untitled](https://github.com/user-attachments/assets/472762e7-da12-40ba-87e3-6ad8b6ff7687)

Reply:
Yes, by just including the token in the header, and it will contain the imagePath that we need to display the user avatar.

Reply:
then , no need for context, should we delete it and just use local storage?

Reply:
Yes, if we don't have something that  we need to be shared globally, then we can delete or comment it for the moment.